src/expr/src/scalar/func/impls/datum.rs

```diff
@@ -21,6 +21,36 @@ sqlfunc!(
     }
 );
 
+sqlfunc!(
+    #[sqlname = "istrue"]
+    #[propagates_nulls = false]
+    #[introduces_nulls = false]
+    #[preserves_uniqueness = false]
+    fn is_true(a: Datum<'_>) -> Result<Option<bool>, EvalError> {
+        Ok(Some(a == Datum::True))
+    }
+);
+
+sqlfunc!(
+    #[sqlname = "isfalse"]
+    #[propagates_nulls = false]
+    #[introduces_nulls = false]
+    #[preserves_uniqueness = false]
+    fn is_false(a: Datum<'_>) -> Result<Option<bool>, EvalError> {
+        Ok(Some(a == Datum::False))
+    }
+);
+
+sqlfunc!(
+    #[sqlname = "isunknown"]
+    #[propagates_nulls = false]
+    #[introduces_nulls = false]
+    #[preserves_uniqueness = false]
+    fn is_unknown(a: Datum<'_>) -> Result<Option<bool>, EvalError> {
+        Ok(Some(a == Datum::Null))
```

Comment:
I think this is correct given the docs here: https://www.postgresql.org/docs/9.1/functions-comparison.html.

Though I'm not sure how to do / if we need to do the "except that the input expression must be of Boolean type"?

Reply:
Yeah, it's a little tricky with these because they're "expressions" and not part of our normal function selection framework.

The gist is that you'll have to make sure that any expression that gets passed to this function has the right `ScalarType`. In most cases that's handled for you, but here we'll need to handroll a type check [here](https://github.com/MaterializeInc/materialize/pull/8455/files#diff-28dcff29ce185a34dc7bb2444006600d245dd969d5bcd3a4e9bef65a36b5efc1R3142).